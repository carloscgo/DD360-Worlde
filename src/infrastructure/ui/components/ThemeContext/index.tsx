import { createContext } from 'react';

type ThemeContextValue = {
    theme: string;
    toggleTheme: (newTheme: string) => void;
}

const ThemeContext = createContext<ThemeContextValue>({
    theme: 'light',
    toggleTheme: () => { },
});

export default ThemeContext;
