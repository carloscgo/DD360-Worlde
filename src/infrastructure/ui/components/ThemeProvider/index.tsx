import { ReactNode, memo, useState } from 'react';
import ThemeContext from '../ThemeContext';
import { Some } from '../../../../application';

type PropsThemeProvider = {
    children: ReactNode
    [key: string]: Some
}

const ThemeProvider = memo(({ children }: PropsThemeProvider) => {
    const [theme, setTheme] = useState('light');

    const toggleTheme = (newTheme: string) => {
        setTheme(newTheme);
    };

    return (
        <ThemeContext.Provider value={{ theme, toggleTheme }}>
            {children}
        </ThemeContext.Provider>
    );
});

export default ThemeProvider;
