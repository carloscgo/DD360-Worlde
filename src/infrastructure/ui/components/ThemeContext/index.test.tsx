import {
    render,
} from '../../../../../setupTest'
import ThemeProvider from '../ThemeProvider';
import ThemeContext from './';

describe('ThemeContext', () => {
    it('renders with default values', () => {
        const { container } = render(<ThemeProvider>
            <ThemeContext.Consumer>{value => JSON.stringify(value)}</ThemeContext.Consumer>
        </ThemeProvider>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders with custom values', () => {
        const theme = 'dark';
        const toggleTheme = jest.fn();
        const values = { theme, toggleTheme }

        const { getByText } = render(
            <ThemeProvider value={values}>
                <ThemeContext.Consumer>
                    {value => (
                        <>
                            <div>{value.theme}</div>
                            <button onClick={() => value.toggleTheme('light')}>Toggle</button>
                        </>
                    )}
                </ThemeContext.Consumer>
            </ThemeProvider>
        );

        const themeElement = getByText('light');
        const toggleButton = getByText('Toggle');

        expect(themeElement).toBeInTheDocument();
        expect(toggleButton).toBeInTheDocument();
    });
});
