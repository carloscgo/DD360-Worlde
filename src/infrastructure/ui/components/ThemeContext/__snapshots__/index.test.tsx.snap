// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeContext renders with default values 1`] = `{"theme":"light"}`;
